src/main/kotlin/org/pkl/intellij/psi/PklModuleNameReference.kt

```diff
@@ -25,15 +26,17 @@ class PklModuleNameReference(private val moduleName: PklModuleName) :
 
   override fun getCanonicalText(): String = moduleName.identifier.text
 
-  override fun resolve(): PklModule? {
+  fun resolveContextual(context: PklProject?): PklModule? {
     val module = moduleName.enclosingModule ?: return null
     val moduleNameText = moduleName.identifier.text
     for (import in module.imports) {
       if (import.memberName == moduleNameText) {
-        val resolved = import.resolve() as? SimpleModuleResolutionResult ?: return null
+        val resolved = import.resolve(context) as? SimpleModuleResolutionResult ?: return null
         return resolved.resolved
       }
     }
     return null
   }
+
+  override fun resolve(): PklModule? = resolveContextual(null)
```

Comment:
This is a method of the plugin SDK's `PsiReference` that we need to implement.

This means that some editor features, like go-to-declaration, can possibly be out-of-sync with our own resolution when we do typechecking and the like.

However, I'm not sure how to better improve this, because we can't just provide new args to `resolve()`. In practice, I think the editing experience is okay (you might see the declaration's version is from package 1.0.1, but during our typecheck it knows it is package 1.1.0).